lib/lsblk/index.ts

```diff
@@ -44,38 +44,50 @@ async function getDevicePaths(): Promise<Map<string, string>> {
 	return mapping;
 }
 
-export async function lsblk(): Promise<Drive[]> {
-	const argv = ['--bytes', '--all'];
-
-	if (SUPPORTS_JSON) {
-		argv.push('--json', '--paths', '--output-all');
-	} else {
-		argv.push('--pairs');
+async function addDevicePaths(devices: Drive[]): Promise<void> {
+	const devicePaths = await getDevicePaths();
+	for (const device of devices) {
+		device.devicePath = devicePaths.get(device.device) || null;
 	}
+}
 
-	let stdout: string;
-	try {
-		[stdout] = await execFile('lsblk', argv);
-	} catch (error) {
-		if (SUPPORTS_JSON) {
+async function getOutput(command: string, ...args: string[]) {
+	const [stdout] = await execFile(command, args);
+	return stdout;
+}
+
+async function lsblkJSON(): Promise<Drive[]> {
+	return parseJSON(
+		await getOutput(
+			'lsblk',
+			'--bytes',
+			'--all',
+			'--json',
+			'--paths',
+			'--output-all',
+		),
+	);
+}
+
+async function lsblkPairs(): Promise<Drive[]> {
+	return parsePairs(await getOutput('lsblk', '--bytes', '--all', '--pairs'));
+}
+
+async function _lsblk(): Promise<Drive[]> {
+	if (SUPPORTS_JSON) {
+		try {
+			return await lsblkJSON();
+		} catch (error) {
```

Comment:
Looks like you could possibly simply this to just:
```javascript
try {
  return await lsblkJSON();
} catch (error) {}
return await lsblkPairs();
```
?

Reply:
@lurch if we know that `lsblk --json` is not supported, we don't want to call it a second time.

Reply:
The current version avoids calling `lsblkJSON` every time when it's unsupported, which probably helps a lot if this function is called regularly (I think etcher polls it? in which case it'd probably make a big difference there)

Reply:
Oh yeah, I forgot that this function gets called repeatedly - sorry!